File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C++/Problem4.cpp
#include <iostream>
#include <cctype>
using namespace std;
int ReadAge() {
	int age;
	cout << "\nEnter your Age : ";
	cin >> age;
	return age;
}

bool HasDriverLicense() {
	string input;
	cout << "\nDo you have a driver license (yes) or (no) : ";
	cin >> input;
	// to convert ToLower
	for (auto &c : input) {
		c = tolower(c);
	}

	if (input == "yes") {
		return true;
	}
	else {
		return false;
	} 
}

void PrintHireOrRejected(bool HasDriverLicense, int age) {
	if (age >= 21 && HasDriverLicense) {
		cout << "\nHired";
	}
	else {
		cout << "\nRejected";
	}
}

int main() {
	PrintHireOrRejected(HasDriverLicense(), ReadAge());
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem4.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        struct stInfo
        {
            public int Age;
            public bool HasDrivingLicense;
        }
        static stInfo ReadInfo()
        {
            stInfo Info = new stInfo();

            Console.Write("\nPlease enter your age : ");
            Info.Age = Int32.Parse(Console.ReadLine());

            Console.Write("\nDo you have a driving license : ");
            Info.HasDrivingLicense = Convert.ToBoolean(Console.ReadLine());

            return Info;
        }
        static bool IsAccepted(stInfo Info)
        {
            return Info.HasDrivingLicense&& Info.Age > 21 ;
        }
        static void PrintResult(stInfo Info)
        {
            if (IsAccepted(Info))
            {
                Console.WriteLine("\nHired");
            }
            else
            {
                Console.WriteLine("\nRejected");
            }
        }
        static void Main(string[] args)
        {
            PrintResult(ReadInfo());
        }
    }
}